pkg/kritis/metadata/containeranalysis/containeranalysis.go

```diff
@@ -107,23 +120,20 @@ func (c Client) fetchVulnerabilityOccurrence(containerImage string, kind string)
 		return nil, fmt.Errorf("%s is not a valid image hosted in GCR", containerImage)
 	}
 
-	req := &grafeas.ListOccurrencesRequest{
-		Filter:   fmt.Sprintf("resourceUrl=%q AND kind=%q", util.GetResourceURL(containerImage), kind),
-		PageSize: constants.PageSize,
-		Parent:   fmt.Sprintf("projects/%s", getProjectFromContainerImage(containerImage)),
-	}
+	req := createListOccurrencesRequest(containerImage, kind)
+	fmt.Println(req)
 
 	it := c.client.ListOccurrences(c.ctx, req)
+	p := iterator.NewPager(it, int(req.PageSize), "")
```

Comment:
What's the reason for doing this differently that in the internal implementation?

Reply:
According to the documentation on iterators (https://github.com/googleapis/google-cloud-go/wiki/Iterator-Guidelines#examples), iteration through pages requires you to use an `iterator.NewPager`. When you call its `NextPage` method, it adds the items from that page to a provided slice. It returns the `nextPageToken`, which will be empty once you have iterated through all the pages.

Reply:
Reverted to original implementation. Integration test shows that there is no bug while extracting vulnerabilities.